cdap-ui/app/cdap/components/Reports/ReportsDetail/Runs/index.js

```diff
@@ -19,53 +19,63 @@ import PropTypes from 'prop-types';
 import {connect} from 'react-redux';
 import {GLOBALS} from 'services/global-constants';
 import {humanReadableDate} from 'services/helpers';
+import {DefaultSelection} from 'components/Reports/store/ActionCreator';
+import difference from 'lodash/difference';
 
 require('./Runs.scss');
 
 const PIPELINES = [GLOBALS.etlDataPipeline, GLOBALS.etlDataStreams];
 
 function getName(run) {
-  let name = run.application.name;
+  if (!run.applicationName) { return '--'; }
 
-  if (PIPELINES.indexOf(run.artifact.name) == -1) {
+  let name = run.applicationName;
+
+  if (PIPELINES.indexOf(run.artifactName) == -1) {
     name = `${name} - ${run.program}`;
   }
 
   return name;
 }
 
 function getType(run) {
-  switch (run.artifact.name) {
+  switch (run.artifactName) {
     case GLOBALS.etlDataPipeline:
       return 'Batch Pipeline';
     case GLOBALS.etlDataStreams:
       return 'Realtime Pipeline';
     default:
-      return run.type;
+      return run.programType;
   }
 }
 
-function renderHeader() {
+function renderHeader(headers) {
   return (
     <div className="grid-header">
       <div className="grid-row">
-        <div>Namespace</div>
-        <div>Name</div>
-        <div>Type</div>
-        <div>Duration</div>
-        <div>Start time</div>
-        <div>End time</div>
-        <div>User</div>
-        <div>Start method</div>
-        <div># Log errors</div>
-        <div># Log warnings</div>
-        <div># Records out</div>
+        <div>
+          Name
+        </div>
+
+        <div>
+          Type
+        </div>
```

Comment:
nit: why not just make these inline divs? Unless you plan to add more attributes